app/models/preprocessor_primo.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# PreprocessorPrimo handles manipulating incoming data from the Primo UI into a structure that TACOS can work with
+class PreprocessorPrimo
+  # to_tacos processes raw incoming query from Primo, looks at each part to see if it is a keyword anywhere search
+  # Any portion that is not a keyword anywhere search drops the entire search from TACOS, logging
+  # as the shared Term `unhandled complex primo query` to allow us to track how frequently we are
+  # dropping terms so we can come back later to build out more complex handing if this is common enough
+  # to warrant the additional work.
+  # @param query [String] example `any,contains,this is a keyword search`
+  def self.to_tacos(query)
+    # split on agreed upon joiner `;;`
+    split_query = query.split(';;')
```

Comment:
Given the filtering that happens before this, I don't think anything needs an adjustment - but I just peeked at our data and there are about a dozen examples of terms already that have `;;` in them. All of these are obviously from not-Primo, and that traffic would not get routed to this model, but it might be a pathway to keep in mind as we investigate anything marked as "unhandled complex primo query"

Reply:
Hmmm. I'll take a closer look and see if there is a better separator to choose that is less likely to exist in an incoming search (I chose this, it's not something Primo provided by default so we can adjust it). Thanks for noticing that.

Reply:
`;;;` seems fine. I've asked AdamS to update the primo integration to use that and will push that change here as well.